crates/turborepo-ui/src/tui/table.rs

```diff
@@ -71,30 +71,54 @@ impl TaskTable {
-            })?;
-        let planned = self.planned.remove(planned_idx);
-        let old_row_idx = self.finished.len() + self.running.len() + planned_idx;
-        let new_row_idx = self.finished.len() + self.running.len();
-        let running = planned.start();
-        self.running.push(running);
+        {
+            let planned = self.planned.remove(planned_idx);
+            let old_row_idx = self.finished.len() + self.running.len() + planned_idx;
+            let new_row_idx = self.finished.len() + self.running.len();
+            let running = planned.start();
+            self.running.push(running);
 
-        if let Some(selected_idx) = self.scroll.selected() {
-            // If task that was just started is selected, then update selection to follow
-            // task
-            if selected_idx == old_row_idx {
-                self.scroll.select(Some(new_row_idx));
-            } else if new_row_idx <= selected_idx && selected_idx < old_row_idx {
-                // If the selected task is between the old and new row positions
-                // then increment the selection index to keep selection the same.
-                self.scroll.select(Some(selected_idx + 1));
+            if let Some(selected_idx) = self.scroll.selected() {
+                // If task that was just started is selected, then update selection to follow
+                // task
+                if selected_idx == old_row_idx {
+                    self.scroll.select(Some(new_row_idx));
+                } else if new_row_idx <= selected_idx && selected_idx < old_row_idx {
+                    // If the selected task is between the old and new row positions
+                    // then increment the selection index to keep selection the same.
+                    self.scroll.select(Some(selected_idx + 1));
+                }
             }
+        } else if let Some(finished_idx) = self
+            .finished
+            .iter()
+            .position(|finished_task| finished_task.name() == task)
+        {
+            let finished = self.finished.remove(finished_idx);
+            let old_row_idx = finished_idx;
+            let new_row_idx = self.finished.len() + self.running.len();
+            let running = finished.start();
+            self.running.push(running);
+
+            if let Some(selected_idx) = self.scroll.selected() {
+                // If task that was just started is selected, then update selection to follow
+                // task
+                if selected_idx == old_row_idx {
+                    self.scroll.select(Some(new_row_idx));
+                } else if new_row_idx <= selected_idx && selected_idx < old_row_idx {
+                    // If the selected task is between the old and new row positions
+                    // then increment the selection index to keep selection the same.
+                    self.scroll.select(Some(selected_idx + 1));
+                }
+            }
+        } else {
```

Comment:
So if a task gets restarted while it's still running it will error?

Reply:
That's currently not possible. We purposefully make it so the same task can't be run at the same time, and we haven't implemented cancelation logic. If a change event happens in the middle of a run, we save it and wait for the run to finish before starting up a new execution.